### PR TITLE
handle scraping failures | define requirements

### PR DESCRIPTION
**Pull request by Kareem-Emad**
This PR is intended to address issues with scraping that makes it stop/throw exception while scraping is in progress. Also I added all the requirements in one file to make it easier to install.

### CODE REVIEW

File: src/dataset_build.py [hunk 1]
```diff
@@ -59,15 +59,15 @@ def create_dataset(query, pics_num, search_engine=SEARCH_ENGINE):
 
 	# Until the number of pictures is not reached
 	while stored < pics_num:
-
-		url = build_url(domain, path, params)
+		if(path):
```
Comment (Sinclert) [inline src/dataset_build.py]:
```suggestion
		if path:
```

File: src/dataset_miner.py [hunk 2]
```diff
@@ -96,13 +96,15 @@ def get_next_page(page, class_name):
 	"""
 
 	urls = page.findAll('a', {'class': class_name})
+	if(len(urls) == 0):
```
Comment (Sinclert) [inline src/dataset_miner.py]:
```suggestion
	if len(urls) == 0:
```

File: src/dataset_miner.py [hunk 3]
```diff
@@ -120,13 +122,20 @@ def get_images(page):
 	"""
 
 	images = page.findAll('img')
-
 	for img in images:
 
-		image_bytes = requests.get(img['src']).content
+		try:
+			if(img.get('src')):
```
Comment (Sinclert) [inline src/dataset_miner.py]:
```suggestion
			if img.get('src'):
```

File: src/video_process.py [hunk 4]
```diff
@@ -46,7 +46,7 @@ def identify_actors(video_path, clf, clf_th, out_name):
 	out_path = compute_path(out_name + '.mp4', 'video')
 	out = cv2.VideoWriter(
 		filename=out_path,
-		fourcc=cv2.VideoWriter_fourcc('X', '2', '6', '4'),
+		fourcc= cv2.VideoWriter_fourcc(*"mp4v"),#cv2.VideoWriter_fourcc('X', '2', '6', '4'),
```
Comment (Sinclert) [inline src/video_process.py]:
```suggestion
		fourcc=cv2.VideoWriter_fourcc(*"mp4v"), # Before: cv2.VideoWriter_fourcc('X', '2', '6', '4'),
```

**Comment by Sinclert** [comment]
Hey Kareem 👋 

Thanks for taking the time to look into this project ^^

I have added some code style suggestion that would make the code more consistent with the rest of the project. Once you apply them, I will approve and merge the PR.

**Comment by Kareem-Emad** [comment]
Thanks for the prompt feedback! I added the changes requested, please check and let me know if you have anymore comments.

**Pull request merged**